core/src/main/java/com/amplitude/core/utilities/Diagnostics.kt

```diff
@@ -15,7 +19,10 @@ class Diagnostics() {
 
     fun addErrorLog(log: String) {
         if (errorLogs == null) {
-            errorLogs = Collections.synchronizedList(mutableListOf())
+            errorLogs = Collections.synchronizedSet(mutableSetOf())
+        }
+        if (errorLogs!!.size >= MAX_ERROR_LOGS) {
+            return
```

Comment:
Should we keep the most recent or least recent error logs on overflow? I think this could go either way, but iOS will keep the most recent and purge older entries and we should make the behavior consistent.

Reply:
I guess either way is fine. I can make it to be the same behavior for consistency

Reply:
Updated